### ISSUE TITLE
Unable to add SAML superadmin attribute 

### ISSUE THREAD

**Issue by nan0viol3t** [ansible/awx#13307]
### Please confirm the following

- [X] I agree to follow this project's [code of conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html).
- [X] I have checked the [current issues](https://github.com/ansible/awx/issues) for duplicates.
- [X] I understand that AWX is open source software provided for free and that I might not receive a timely response.

### Bug Summary

I'm trying to add SAML User Flags Attribute Mapping for AWX instance admin / superadmin based on AD role,
based on documentation -> https://github.com/ansible/awx/blob/devel/docs/auth/saml.md#example-saml-user-flags-attribute-mapping

entry:
```
{
  "is_superuser_role": "my_ad_admin_role_name"
}

```
but this throws an error:
> Expected a list of items but got type "str"

I tried many variations of such entry, without success, with different errors of syntax.
I spent hours (days actually) for potential correct syntax of this setting and still unable to grant instance admin privileges based on SAML role (same story for "attr" or "value").
Can you provide example syntaxt, as I'm convinced for now, that this is a bug.
I will gladly perform tests with your guidance, as I have no more ideas as of now.
 

### AWX version

21.8.0

### Select the relevant components

- [X] UI
- [ ] API
- [ ] Docs
- [ ] Collection
- [ ] CLI
- [X] Other

### Installation method

kubernetes

### Modifications

yes

### Ansible version

2.12.5

### Operating system

Amazon Linux 2

### Web browser

Firefox

### Steps to reproduce

Setup of SAML User Flags Attribute Mapping with one of below attribute in AWX SAML configuration tab - :

{
  "is_superuser_role": "awx_admins",
  "is_superuser_attr": "is_superuser",
  "is_superuser_value": "IT-Superadmin",
  "is_system_auditor_role": "awx_auditors",
  "is_system_auditor_attr": "is_system_auditor",
  "is_system_auditor_value": "Auditor"
}

### Expected results

Syntax error of different kinds. Missing proper declaration. 

### Actual results

Failing to save specific attribute due to unknown syntax scheme.

**Comment by nan0viol3t** [comment]
Probably duplicated to this https://github.com/ansible/awx/issues/13049#issue-1406466139 but I'm convinced that this is not only documentation problem.

**Comment by nan0viol3t** [comment]
...and to my previous comment and https://github.com/ansible/awx/issues/13049#issue-1406466139 mentioned solution:

> step 17 here describes it correctly https://docs.ansible.com/automation-controller/latest/html/administration/ent_auth.html#saml-settings

this also does not work, as in form presented on that page, step 17, AWX do not allow to save attr list due to error:

> "Expected a list of items but got type "str"."

**Comment by john-westcott-iv** [comment]
According to item 17 from that linked doc role and value are now "one or more" fields. So they both need to be arrays. Based on the config you posted above, could you please try the following and let us know if it works?
```
{
"is_superuser_role": ["awx_admins"],
"is_superuser_attr": "is_superuser",
"is_superuser_value": ["IT-Superadmin"],
"is_system_auditor_role": ["awx_auditors"],
"is_system_auditor_attr": "is_system_auditor",
"is_system_auditor_value": ["Auditor"]
}
```

If it still fails, I might try to add one item at a time to see what setting hits the issue.

**Comment by nan0viol3t** [comment]
Appreciate response and help John.
Indeed, I manage to save SAML option with "array" syntax.
Unfortunately, permission declaration is not working, mean:

{ 
"is_superuser_role": ["my_AD_group_name"]
}

does not grant me AWX instance admin permissions/admin account type.

I will test some more SAML attribute configurations in next days, once I'll address another issues with current platform. I'll post any update in here, just with some delay... and apologize in advance for delayed response.

**Comment by nan0viol3t** [comment]
I'm back,
apologize for very long silence. I wrote before I'll response with delay, but I was not expecting such long period...
Back to SAML topic, I feel I'm missing something and it might be related to Azure and our dev app configuration, but having role created/assigned, I'm still unable to get super user permissions based on SAML auth.

As previously, my user att flag looks like this:
(obfuscated names)

```
{
"is_superuser_role": ["AWX_Role_1"]
}
```

and this is what I can see in debug mode:
(modified syntax due to sensitive data)

```
'http://schemas.xmlsoap.org/ws/2005/05/identity/claims/name': ['Obfuscated1'], 'Roles': ['AWX_Role_1', 'AWX_Role_2', 'AWX_Role_3'], 'name_id': 'Obfuscated1'} 
2023-01-20 00:10:27,670 DEBUG  awx.sso.pipeline User Obfuscated1 is missing the superuser role(s) AWX_Role_1
2023-01-20 00:10:27,679 DEBUG  awx.sso.backends Enterprise user Obfuscated1 already created in Tower.
2023-01-20 00:10:27,709 DEBUG  awx.analytics.performance request: <WSGIRequest: POST '/sso/complete/saml/'>, response_time: 0.151s                                     

```

// So, I'm able to login using SAML, but still missing super user permission/account type.
From logs, it looks like I have necessary role "AWX_Role_1", but next log lines says, that I'm missing super user role "AWX_Role_1"
Is this something recognized before? Maybe I shouldnt use underscore in roles naming convention?

Probably I'm missing something but not sure what exactly, can you help ?

**Comment by nan0viol3t** [comment]
Update

I was able to setup SAML auth and instance admin permissions, by adding user attr on Azure App group level and then on AWX like:
```
{
"is_superuser_attr": "is_superuser",
"is_superuser_value": ["MYATTR"]
}
```

I can confirm that this is working, 
but using only group membership and then using role attr (what I was initially doing, like in previous post):
```
{
"is_superuser_role": ["AWX_Role_1"]
}
```
IS not working, 
and I can see in AWX logs that it correctly recognize my AD groups and roles which comes from group membership, but still cant grant me superuser rights, as it says I have no required role, like in my previous comment, it shows I have necessary role in one log line and in second log entry, it says I do not have necessary role to have superuser perms...
I wonder if this is something related to var type which is passed in time of "is_superuser_role" read.
Not sure if this is a bug or still something on mu Azure APP configuration - but looking at logs entry I would say, AWX is not recognizing roles correctly in time of reading "is_superuser_role" var.

Fortunately, with superuser_attr and value, I was able to achieve expected behavior and now I have superuser priv, once logged via SAML.
Hope those comments brings any value to others.
Not sure if this can be closed but my problem is solved.
Leaving this open for now - maybe someone else will comment or advise closing...
